app/assets/javascripts/application.js.erb

```diff
@@ -6,245 +6,248 @@
 //
 // It's not advisable to add code directly here, but if you do, it'll appear at the bottom of the
 // compiled file.
-//
+// 
 // Read Sprockets README (https://github.com/sstephenson/sprockets#sprockets-directives) for details
 // about supported directives.
-//
-//= require jquery
+// 
+//= require jquery 
 //= require jquery_ujs
-//= require bootstrap-sprockets
-//= require dataTables/jquery.dataTables
+//= require bootstrap-sprockets 
+//= require dataTables/jquery.dataTables  
 //= require dataTables/bootstrap/3/jquery.dataTables.bootstrap
-//= require lightbox2/javascripts/lightbox
-//= require oboe/oboe-browser.min
-//= require datatables_plugins/api/processing
+//= require lightbox2/javascripts/lightbox 
+//= require ndjson/ndjsonStream   
+//= require oboe/oboe-browser.min  
+//= require datatables_plugins/api/processing 
 //= require_tree .
 
-function fetch_table_data(table, options){
-  if (!options) options = {};
-  if (!options.doneCallback) options.doneCallback = null;
-  if (!options.base_uri) options.base_uri = window.location.pathname;
-
-  oboe({
-    url: options.base_uri + '/jobs.json?'+get_request_params(),
-    headers: {
-      'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
-      'X-Requested-With': 'XMLHttpRequest',
-      'Content-Type': 'application/json'
-    }
-  }).start(function(){
-    table.processing(true);
-  }).node('data.*', function(jobs){
-    // data is an array of arrays, where each array has one or more jobs
-    // this is called once for each array of jobs
-    table.rows.add(jobs).draw();
-    table.processing(false);
-  }).node('errors.*', function(error){
-    // errors is a single array of error messages
-    // this is called once for each error msg
-    show_errors([error]);
-  }).done(function(){
-    table.processing(false);
+function fetch_table_data(table, options) {
+    if (!options) options = {};
+    if (!options.doneCallback) options.doneCallback = null;
+    if (!options.base_uri) options.base_uri = window.location.pathname;
 
-    if(options.doneCallback){
-       options.doneCallback();
-    }
-  }).fail(function(errorReport){
-    if(errorReport.statusCode != null){
-      show_errors(["Request for jobs failed with status code: " + errorReport.statusCode]);
-    }
-    else{
-      //FIXME: this error appears even when the above 404 occurs, for example
-      // that is because a 404 responce for json request returns a plain text response
-      // and parsing that as json fails
-       show_errors(["Request for jobs failed due to body parsing error."])
-    }
+    fetch(options.base_uri + '/jobs.json?' + get_request_params(), {
+            headers: {
+                'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
+                'X-Requested-With': 'XMLHttpRequest',
+                'Content-Type': 'application/json'
+            }
+        }).catch(e => {
+            console.log(e)
```

Comment:
in the original oboe.js code there was one "fail" block at the end that handled the request errors, and we handled the request errors by actually calling `show_errors` function which will display to the user.

Here and below you have added catch blocks that instead write an error to the console.log but from the user's perspective fail silently. What exceptions are you catching here, can they be caught at the end, and do you know what error message you would display.

Also, in exception cases, `table.processing(false)` should be called to turn the indicator off.